presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaRecordSet.java

```diff
@@ -84,11 +85,54 @@
 
         ImmutableList.Builder<Type> typeBuilder = ImmutableList.builder();
 
-        for (DecoderColumnHandle handle : columnHandles) {
+        for (DecoderColumnHandle handle : this.columnHandles) {
             typeBuilder.add(handle.getType());
         }
 
         this.columnTypes = typeBuilder.build();
+        KafkaConsumer consumer = consumerManager.createConsumer(Thread.currentThread().getId(), split.getLeader());
+        TopicPartition tp = new TopicPartition(split.getTopicName(), split.getPartitionId());
+        consumer.assign(ImmutableList.of(tp));
+        setOffsetRange(consumer, split);
+        consumer.close();
+    }
+
+    private static void setOffsetRange(KafkaConsumer<Long, String> consumer, KafkaSplit split)
+    {
+        if (split.getStartTs() > split.getEndTs()) {
+            throw new IllegalArgumentException(String.format("Invalid Kafka Offset start/end pair: %s - %s", split.getStartTs(), split.getEndTs()));
+        }
+
+        TopicPartition topicPartition = new TopicPartition(split.getTopicName(), split.getPartitionId());
+        consumer.assign(ImmutableList.of(topicPartition));
+
+        long beginningOffset =
+                (split.getStartTs() == 0) ?
+                        consumer.beginningOffsets(ImmutableList.of(topicPartition)).values().iterator().next() :
+                        findOffsetsByTimestamp(consumer, topicPartition, split.getStartTs());
+        long endOffset =
+                (split.getEndTs() == 0) ?
+                        consumer.endOffsets(ImmutableList.of(topicPartition)).values().iterator().next() :
+                        findOffsetsByTimestamp(consumer, topicPartition, split.getEndTs());
+
+        split.setStart(beginningOffset);
+        split.setEnd(endOffset);
+    }
+
+    private static long findOffsetsByTimestamp(KafkaConsumer<Long, String> consumer, TopicPartition tp, long timestamp)
```

Comment:
please not use variable abbreviation:
s/tp/topicPartition/g
s/tpOffsets/topicPartitionOffsets/g

Reply:
Fixed.